shard-operator-parent/shard-operator-core/src/test/java/com/redhat/service/smartevents/shard/operator/core/observability/ObservabilitySetupServiceTest.java

```diff
@@ -0,0 +1,194 @@
+import io.fabric8.kubernetes.api.model.ObjectMeta;
+import io.fabric8.kubernetes.api.model.Secret;
+import io.fabric8.kubernetes.api.model.SecretList;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.client.dsl.MixedOperation;
+import io.fabric8.kubernetes.client.dsl.NonNamespaceOperation;
+import io.fabric8.kubernetes.client.dsl.Resource;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoInteractions;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+public class ObservabilitySetupServiceTest {
+
+    private static String NAME = "name";
+    private static String NAMESPACE = "namespace";
+    private static String ACCESS_TOKEN = "access-token";
+    private static String REPOSITORY = "repository";
+    private static String CHANNEL = "channel";
+    private static String TAG = "tag";
+
+    private static Base64.Encoder ENCODER = Base64.getEncoder();
+
+    @Mock
+    KubernetesClient client;
+
+    @Mock
+    TemplateProvider templateProvider;
+
+    @Mock
+    MixedOperation<Secret, SecretList, Resource<Secret>> secrets;
+
+    @Mock
+    NonNamespaceOperation<Secret, SecretList, Resource<Secret>> secretsNonNamespaceOperation;
+
+    @Mock
+    Resource<Secret> secretResource;
+
+    @Mock
+    NonNamespaceOperation<Namespace, NamespaceList, Resource<Namespace>> namespaceNonNamespaceOperation;
+
+    @Mock
+    Resource<Namespace> namespaceResource;
+
+    ObservabilitySetupServiceForTests handler;
+
+    @BeforeEach
+    void setup() {
+        this.handler = new ObservabilitySetupServiceForTests(client, templateProvider);
+    }
+
+    @ParameterizedTest()
+    @MethodSource("parameters")
+    void testMissingParameter(String name, String namespace, String accessToken, String repository, String channel, String tag) {
+        handler.setEnabled(true);
+
+        handler.setName(name);
+        handler.setNamespace(namespace);
+        handler.setAccessToken(accessToken);
+        handler.setRepository(repository);
+        handler.setChannel(channel);
+        handler.setTag(tag);
+
+        handler.createOrUpdateObservabilitySecret();
+
+        verifyNoInteractions(client);
+    }
+
+    private static Stream<Arguments> parameters() {
+        Object[][] parameters = {
+                { null, NAMESPACE, ACCESS_TOKEN, REPOSITORY, CHANNEL, TAG },
+                { NAME, null, ACCESS_TOKEN, REPOSITORY, CHANNEL, TAG },
+                { NAME, NAMESPACE, null, REPOSITORY, CHANNEL, TAG },
+                { NAME, NAMESPACE, ACCESS_TOKEN, null, CHANNEL, TAG },
+                { NAME, NAMESPACE, ACCESS_TOKEN, REPOSITORY, null, TAG },
+                { NAME, NAMESPACE, ACCESS_TOKEN, REPOSITORY, CHANNEL, null }
+        };
+        return Stream.of(parameters).map(Arguments::of);
+    }
+
```

Comment:
@manstis Perhaps one more test just to verify that things work as expected when it's disabled?

Reply:
Added.